Composer/packages/client/src/components/Header.tsx

```diff
@@ -95,33 +121,110 @@ export const Header = () => {
   const locale = useRecoilValue(localeState(projectId));
   const appUpdate = useRecoilValue(appUpdateState);
   const { showing, status } = appUpdate;
+  const [showStartBotsPanel, setStartPanelView] = useState(false);
+  const runningBots = useRecoilValue(runningBotsSelector);
+  const [startPanelText, setStartPanelText] = useState('');
+  const { projectIds: projectIdsRunning } = runningBots;
+  const [areBotsStarted, setBotsInBotProjectStarted] = useState<boolean>(false);
+  const { stopAllBots, startAllBots } = useLocalBotOperations();
+  const [showStartBotsWidget, setStartBotsWidgetVisible] = useState(true);
+  const currentMode = useRecoilValue(currentModeState);
+
+  useEffect(() => {
+    if (currentMode !== 'home') {
+      setStartBotsWidgetVisible(true);
+      return;
+    }
+    setStartBotsWidgetVisible(false);
+  }, [currentMode]);
+
+  useEffect(() => {
+    if (projectIdsRunning.length > 0) {
+      setBotsInBotProjectStarted(true);
+    }
+  }, [projectIdsRunning]);
+
+  useEffect(() => {
+    if (runningBots.projectIds.length > 0) {
+      setStartPanelText(
+        `${formatMessage('Stop all bots')} (${runningBots.projectIds.length}/${runningBots.totalBots} ${formatMessage(
+          'running'
+        )})`
+      );
+    } else {
+      setStartPanelText(formatMessage('Start all bots'));
+    }
+  }, [runningBots]);
```

Comment:
You should use the `useMemo` hook instead of `useState` and `useEffect` where you can. 

```js
 const buttonText = useMemo(() => {
    if (running) {
      return formatMessage('Stop all bots ({running}/{total}) running', {
        running: runningBots.projectIds.length,
        total: runningBots.totalBots,
      });
    }
    return formatMessage('Start all bots');
  }, [runningBots, running]);
```